elizaOS/eliza
# feat: add avail plugin

# Relates to:

N/A



# Risks

Low.
The main implementation risk is that the Plugin doesn't work as intended.

# Background

## What does this PR do?

[Avail](https://www.availproject.org/) aims to accelerate Web3's unification using its cutting-edge modular technology stack, which combines data availability, aggregation, and shared security.

In this version, the `transfer` and `submitData` actions are supported.

Multiple L2s are working with Avail to meet their data availability and interoperability needs.

## What kind of change is this?

Features (non-breaking change which adds functionality)




# Documentation changes needed?
Since Avail is a Substrate based network, and not EVM-compatible (cannot work with `plugin-evm`), it requires some docs to show how to submit data and transfer tokens.  I have updated the documentation accordingly in the README for the plugin.




# Testing

## Where should a reviewer start?

Most of the changes are in `plugin-avail`


## Detailed testing steps
- In `.env` you should set the value for `AVAIL_ADDRESS` (this is the public address for the agent account) and `AVAIL_SEED` (seed phrase for the same account).
- **Transfer AVAIL**
    - To test transfer function, you need tokens in your Avail account. On testnet, you can use the [Avail Faucet](https://faucet.avail.tools/). If you need more please ping me, and I can send it over. 
    - Run the agent and prompt it with: "send 1 AVAIL to <any other Avail account> " - e.g. `send 1 AVAIL to 5GWbvXjefEvXXETtKQH7YBsUaPc379KAQATW1eqeJT26cbsK`
    - Assuming everything goes smoothly the agent returns with the Tx Hash, and Block Hash. 
The tx hash can be checked on the Avail block explorer at https://avail-turing.subscan.io/
- **Submit Data**
    - To test data submission,  you need tokens in your Avail account to pay fees. On testnet, you can use the [Avail Faucet](https://faucet.avail.tools/). If you need more please ping me, and I can send it over. 
    - Run the agent and prompt it with: "Submit the following data to Avail <DATA>" - e.g. `Submit the following data to Avail "Hello World!"`
    - Assuming everything goes smoothly the agent returns with the Tx Hash, and Block Hash. 

















## Discord username
RobinRRT
